packages/keyring/src/keyring.ts

```diff
@@ -85,9 +85,11 @@ export default class Keyring implements KeyringInstance {
    * `addPair` to stores in a keyring pair dictionary the public key of the generated pair as a key and the pair as the associated value.
    */
   public addFromAddress (address: string | Uint8Array, meta: KeyringPair$Meta = {}, encoded: Uint8Array | null = null, type: KeypairType = this.type, ignoreChecksum?: boolean): KeyringPair {
-    const publicKey = this.decodeAddress(address, ignoreChecksum);
+    address = this.decodeAddress(address, ignoreChecksum);
+    const publicKey = type === 'ecdsa'
+                    ? new Uint8Array() : address;
```

Comment:
Always pass the publicKey, don't handle it differently. Since here you are using the address as the public identifier anyway, pass it as such.